framework/src/parser/Parser.C

```diff
@@ -530,17 +530,33 @@ Parser::buildJsonSyntaxTree(JsonSyntaxTree & root) const
     if (act_info._task == "")
       act_info._task = _action_factory.getTaskName(act_info._action);
 
-    all_names.push_back(std::pair<std::string, Syntax::ActionInfo>(iter.first, act_info));
+    all_names.push_back(std::make_pair(iter.first, act_info));
   }
 
   for (const auto & act_names : all_names)
   {
-    InputParameters action_obj_params = _action_factory.getValidParams(act_names.second._action);
-    root.addParameters("", act_names.first, false, act_names.second._action, act_names.second._task,
-                       true, &action_obj_params);
-
-    const std::string & task = act_names.second._task;
-    std::string act_name = act_names.first;
+    const auto & act_info = act_names.second;
+    const std::string & action = act_info._action;
+    const std::string & task = act_info._task;
+    const std::string act_name = act_names.first;
+    InputParameters action_obj_params = _action_factory.getValidParams(action);
+    bool params_added = root.addParameters("",
+                                           act_name,
+                                           false,
+                                           action,
+                                           true,
+                                           &action_obj_params,
+                                           _syntax.getLineInfo(act_name, action, ""));
```

Comment:
Also, there is another call to addParameters below this that doesn't use information from act_info.